qutebrowser/mainwindow/mainwindow.py

```diff
@@ -127,6 +127,45 @@ def get_target_window():
         return None
 
 
+def _generate_stylesheet_statusbar():
+    flags = [
+        ('private', 'statusbar.private'),
+        ('caret', 'statusbar.caret'),
+        ('caret-selection', 'statusbar.caret.selection'),
+        ('prompt', 'prompts'),
+        ('insert', 'statusbar.insert'),
+        ('command', 'statusbar.command'),
+        ('passthrough', 'statusbar.passthrough'),
+        ('private-command', 'statusbar.command.private'),
+    ]
+    stylesheet = """
+        StatusBar,
+        StatusBar QLabel,
+        StatusBar QLineEdit {
+            font: {{ conf.fonts.statusbar }};
+            color: {{ conf.colors.statusbar.normal.fg }};
+        }
+
+        StatusBar {
+            background-color: {{ conf.colors.statusbar.normal.bg }};
+        }
+    """
+    for flag, option in flags:
+        stylesheet += """
+            StatusBar[color_flags~="%s"],
+            StatusBar[color_flags~="%s"] QLabel,
+            StatusBar[color_flags~="%s"] QLineEdit {
+                color: {{ conf.colors.%s }};
+            }
+
+            StatusBar[color_flags~="%s"] {
+                background-color: {{ conf.colors.%s }};
+            }
+        """ % (flag, flag, flag,  # noqa: S001
```

Comment:
What is `S001`?

Reply:
It's coming from [flake8-pep3101](https://github.com/gforcada/flake8-pep3101) and saying the same thing you said above: Use `.format()` instead of percent-formatting :wink:

Reply:
That comes from old code.

However, as this solution may (I just think so without benchmarking) degrades performance after #4808 is merged, I don't think it's a good idea anymore. (see https://github.com/qutebrowser/qutebrowser/issues/4686#issuecomment-495982864-permalink)

Reply:
Ah, right, my bad - I used percent-formatting there because the string already uses `{` (for the CSS-like syntax) and `{{` (for jinja), so doing that was much more readable than using `{{` and `{{{{` everywhere.